### PR TITLE
Create new exceptions

### LINKED ISSUE #772: NotImplementedException to go with Response.Status.NOT_IMPLEMENTED & friends
Provide equivalent exception types for the new v2.0 Response.Status fields.

### PR DESCRIPTION
**Pull request by jeyvison**
Closes #772 

See https://github.com/eclipse-ee4j/jaxrs-api/pull/774

### CODE REVIEW

File: jaxrs-api/src/main/java/jakarta/ws/rs/BadGatewayException.java [hunk 1]
```diff
@@ -0,0 +1,105 @@
+/********************************************************************************
+ * Copyright (c) 2019 Jeyvison Nascimento
```
Comment (andymc12) [inline jaxrs-api/src/main/java/jakarta/ws/rs/BadGatewayException.java:2]:
```suggestion
 * Copyright (c) 2020 Jeyvison Nascimento
```

File: jaxrs-api/src/main/java/jakarta/ws/rs/ExpectationFailedException.java [hunk 2]
```diff
@@ -0,0 +1,104 @@
+/********************************************************************************
+ * Copyright (c) 2019 Jeyvison Nascimento
```
Comment (andymc12) [inline jaxrs-api/src/main/java/jakarta/ws/rs/ExpectationFailedException.java:2]:
```suggestion
 * Copyright (c) 2020 Jeyvison Nascimento
```

File: jaxrs-api/src/main/java/jakarta/ws/rs/GatewayTimeoutException.java [hunk 3]
```diff
@@ -0,0 +1,105 @@
+/********************************************************************************
+ * Copyright (c) 2019 Jeyvison Nascimento
```
Comment (andymc12) [inline jaxrs-api/src/main/java/jakarta/ws/rs/GatewayTimeoutException.java:2]:
```suggestion
 * Copyright (c) 2020 Jeyvison Nascimento
```

File: jaxrs-api/src/main/java/jakarta/ws/rs/HttpVersionNotSupportedException.java [hunk 4]
```diff
@@ -0,0 +1,105 @@
+/********************************************************************************
+ * Copyright (c) 2019 Jeyvison Nascimento
```
Comment (andymc12) [inline jaxrs-api/src/main/java/jakarta/ws/rs/HttpVersionNotSupportedException.java:2]:
```suggestion
 * Copyright (c) 2020 Jeyvison Nascimento
```

File: jaxrs-api/src/main/java/jakarta/ws/rs/LengthRequiredException.java [hunk 5]
```diff
@@ -0,0 +1,106 @@
+/********************************************************************************
+ * Copyright (c) 2019 Jeyvison Nascimento
```
Comment (andymc12) [inline jaxrs-api/src/main/java/jakarta/ws/rs/LengthRequiredException.java:2]:
```suggestion
 * Copyright (c) 2020 Jeyvison Nascimento
```

File: jaxrs-api/src/main/java/jakarta/ws/rs/MethodNotAllowedException.java [hunk 6]
```diff
@@ -0,0 +1,107 @@
+/********************************************************************************
+ * Copyright (c) 2019 Jeyvison Nascimento
```
Comment (andymc12) [inline jaxrs-api/src/main/java/jakarta/ws/rs/MethodNotAllowedException.java:2]:
```suggestion
 * Copyright (c) 2020 Jeyvison Nascimento
```

File: jaxrs-api/src/main/java/jakarta/ws/rs/NetworkAuthenticationRequiredException.java [hunk 7]
```diff
@@ -0,0 +1,104 @@
+/********************************************************************************
+ * Copyright (c) 2019 Jeyvison Nascimento
```
Comment (andymc12) [inline jaxrs-api/src/main/java/jakarta/ws/rs/NetworkAuthenticationRequiredException.java:2]:
```suggestion
 * Copyright (c) 2020 Jeyvison Nascimento
```

File: jaxrs-api/src/main/java/jakarta/ws/rs/PreconditionRequiredException.java [hunk 8]
```diff
@@ -0,0 +1,104 @@
+/********************************************************************************
+ * Copyright (c) 2019 Jeyvison Nascimento
```
Comment (andymc12) [inline jaxrs-api/src/main/java/jakarta/ws/rs/PreconditionRequiredException.java:2]:
```suggestion
 * Copyright (c) 2020 Jeyvison Nascimento
```

File: jaxrs-api/src/main/java/jakarta/ws/rs/ProxyAutheticationRequiredException.java [hunk 9]
```diff
@@ -0,0 +1,106 @@
+/********************************************************************************
+ * Copyright (c) 2019 Jeyvison Nascimento
```
Comment (andymc12) [inline jaxrs-api/src/main/java/jakarta/ws/rs/ProxyAutheticationRequiredException.java:2]:
```suggestion
 * Copyright (c) 2020 Jeyvison Nascimento
```

File: jaxrs-api/src/main/java/jakarta/ws/rs/RequestEntityTooLargeException.java [hunk 10]
```diff
@@ -0,0 +1,106 @@
+/********************************************************************************
+ * Copyright (c) 2019 Jeyvison Nascimento
```
Comment (andymc12) [inline jaxrs-api/src/main/java/jakarta/ws/rs/RequestEntityTooLargeException.java:2]:
```suggestion
 * Copyright (c) 2020 Jeyvison Nascimento
```

File: jaxrs-api/src/main/java/jakarta/ws/rs/RequestHeaderFieldsTooLargeException.java [hunk 11]
```diff
@@ -0,0 +1,105 @@
+/********************************************************************************
+ * Copyright (c) 2019 Jeyvison Nascimento
```
Comment (andymc12) [inline jaxrs-api/src/main/java/jakarta/ws/rs/RequestHeaderFieldsTooLargeException.java:2]:
```suggestion
 * Copyright (c) 2020 Jeyvison Nascimento
```

File: jaxrs-api/src/main/java/jakarta/ws/rs/RequestTimeoutException.java [hunk 12]
```diff
@@ -0,0 +1,107 @@
+/********************************************************************************
+ * Copyright (c) 2019 Jeyvison Nascimento
```
Comment (andymc12) [inline jaxrs-api/src/main/java/jakarta/ws/rs/RequestTimeoutException.java:2]:
```suggestion
 * Copyright (c) 2020 Jeyvison Nascimento
```

File: jaxrs-api/src/main/java/jakarta/ws/rs/RequestUriTooLong.java [hunk 13]
```diff
@@ -0,0 +1,105 @@
+/********************************************************************************
+ * Copyright (c) 2019 Jeyvison Nascimento
```
Comment (andymc12) [inline jaxrs-api/src/main/java/jakarta/ws/rs/RequestUriTooLong.java:2]:
```suggestion
 * Copyright (c) 2020 Jeyvison Nascimento
```

File: jaxrs-api/src/main/java/jakarta/ws/rs/RequestedRangeNotSatisfiableException.java [hunk 14]
```diff
@@ -0,0 +1,104 @@
+/********************************************************************************
+ * Copyright (c) 2019 Jeyvison Nascimento
```
Comment (andymc12) [inline jaxrs-api/src/main/java/jakarta/ws/rs/RequestedRangeNotSatisfiableException.java:2]:
```suggestion
 * Copyright (c) 2020 Jeyvison Nascimento
```

File: jaxrs-api/src/main/java/jakarta/ws/rs/TooManyRequestsException.java [hunk 15]
```diff
@@ -0,0 +1,105 @@
+/********************************************************************************
+ * Copyright (c) 2019 Jeyvison Nascimento
```
Comment (andymc12) [inline jaxrs-api/src/main/java/jakarta/ws/rs/TooManyRequestsException.java:2]:
```suggestion
 * Copyright (c) 2020 Jeyvison Nascimento
```

**Review by andymc12 — CHANGES_REQUESTED** [review]
Hi @jeyvison - overall, looks good.  A few comments:
Copyright should be 2020.
Also, could you target the PR to `master` branch now instead of `3.1-SNAPSHOT`?
Thanks!

**Comment by jeyvison** [comment]
Done. See https://github.com/eclipse-ee4j/jaxrs-api/pull/774

**Pull request closed without merging**